airflow/configuration.py

```diff
@@ -267,17 +271,49 @@ def _get_env_var_option(self, section, key):
         env_var_cmd = env_var + '_CMD'
         if env_var_cmd in os.environ:
             # if this is a valid command key...
-            if (section, key) in self.as_command_stdout:
+            if (section, key) in self.configs_with_secret:
                 return run_command(os.environ[env_var_cmd])
 
     def _get_cmd_option(self, section, key):
         fallback_key = key + '_cmd'
         # if this is a valid command key...
-        if (section, key) in self.as_command_stdout:
+        if (section, key) in self.configs_with_secret:
             if super().has_option(section, fallback_key):
                 command = super().get(section, fallback_key)
                 return run_command(command)
 
+    @cached_property
+    def _secrets_backend_client(self):
+        if super().has_option('secrets', 'backend') is False:
+            return None
+
+        secrets_backend_cls = super().get('secrets', 'backend')
+        if not secrets_backend_cls:
+            return None
+        print("secrets_backend_cls", secrets_backend_cls)
+        secrets_backend = import_string(secrets_backend_cls)
+        if secrets_backend:
+            try:
+                alternative_secrets_config_dict = json.loads(
+                    super().get('secrets', 'backend_kwargs', fallback='{}')
+                )
+            except json.JSONDecodeError:
+                alternative_secrets_config_dict = {}
+            return secrets_backend(**alternative_secrets_config_dict)
```

Comment:
Hmmmm, not a huge fan of the duplication here and in `airflow/secrets/__init__.py`.

Since this would only be called _after_ the class is constructed would we instead do


```suggestion
    @cached_property
    def _secrets_backend_client(self):
        if super().has_option('secrets', 'backend') is False:
            return None

        from airflow.secrets import secrets_backend_list
        return secrets_backend_list
```

(give or take).

By delaying the import to inside the function no import loop is created.

This probably also means adding a `get_config(section, key)` to airflow.secrets, and possibly changing the default implemenation from `NotImplementedError` to `return None`.

Note: we should pass they key and section in directly, and let the secret backends convert that to a single parameter _if needed_ (for instance we could have /config/core/sql_alchemy_conn, rather than config/core__sql_alchemy_conn)

Reply:
The main there is some code duplication here is we can't use `airflow.secrets` as otherwise we would have infinite loop since the DEFAULT_SECRET_BACKEND ( Environment Variable and Connection ) needs `sql_achemy_conn` value

Reply:
```ini
sql_alchemy_conn_secret = conn/sql_alchemy_conn
```
should already allow that.

Whatever they pass to `sql_alchemy_conn_secret` will be directly used to get secret

Reply:
Updated with TYPE_CHECKING